ai-decision-services/ai-ad-dpp-engine/prepare-config-data-and-app/prepare-config-data-and-app.md

```diff
@@ -0,0 +1,217 @@
+
+Lab 3: Preparation code, config, library and datasets
+===
+
+Fundamentally, you will need the following things ready for setting up the whole infrastructure:
+
+*   Orchestration configuration
+*   Driver code (Orchestrator) - from our repo
+*   Training/Inferencing datasets
+*   Library (Archive.zip)
+
+## Orchestration Configuration
+
+This is a JSON file that defines the DPP workflow. The configuration is composed of the following 5 parts:
+
+<table class="wrapped confluenceTable"><colgroup><col><col></colgroup><tbody><tr><th class="confluenceTh">Section Name</th><th class="confluenceTh">Explanation</th></tr><tr><td class="confluenceTd">inputSources</td><td class="confluenceTd">Connector for input data sources. Currently we support reading data from Object Storage, ATP and ADW.</td></tr><tr><td class="highlight-blue confluenceTd" data-highlight-colour="blue">phaseInfo<td class="highlight-blue confluenceTd" data-highlight-colour="blue">A flag specifies whether it's training or inferencing, and connector to metadata source.</td></tr><tr><td class="confluenceTd">processingSteps</td><td class="confluenceTd">Data preprocessing transformers and related input arguments.</td></tr><tr><td class="highlight-blue confluenceTd" data-highlight-colour="blue">stagingDestination</td><td class="highlight-blue confluenceTd" data-highlight-colour="blue">Connector for intermediate processed data storage, which will be used in post-processing steps.</td></tr><tr><td class="confluenceTd">outputDestination</td><td class="confluenceTd">Connector for final output.</td></tr><tr><td class="highlight-blue confluenceTd" data-highlight-colour="blue">serviceApiConfiguration (Optional)</td><td class="highlight-blue confluenceTd" data-highlight-colour="blue">Configuration required for post-processing steps. For example, arguments for anomaly detection client.</td></tr></tbody></table>
+
+## inputSources
+
+Dataset(s) input source(s), with **inputSources** as key and an array of dataset(s) information as value.
```

Comment:
This sentence does not make sense - 'The input data source configuration', shouldn't this be sufficient?